ui/src/app/components/help-sidebar-icons.tsx

```diff
@@ -113,8 +127,58 @@ export interface IconConfig {
   hasContent: boolean;
 }
 
+export interface StatusIconProps {
+  iconPath: string;
+  iconConfig: IconConfig;
+  config: ConfigResponse;
+  children: React.ReactNode;
+}
+
+export const StatusIcon = ({
+  iconPath,
+  iconConfig,
+  config,
+  children,
+}: StatusIconProps) => {
+  const iconStyle: CSSProperties = showAppsPanel(config)
+    ? { width: '36px', position: 'absolute' }
+    : { width: '22px', position: 'absolute' };
+
+  return (
+    <FlexRow style={containerStyle}>
+      <img
+        src={iconPath}
+        alt={iconConfig.label}
+        style={iconStyle}
+        data-test-id={'help-sidebar-icon-' + iconConfig.id}
+      />
+      {children}
+    </FlexRow>
+  );
+};
+
+const displayAppStatusIcon = (
+  iconConfig: IconConfig,
+  workspaceNamespace: string,
+  userSuspended: boolean,
+  config: ConfigResponse,
+  status: AppStatus,
+  appType: UIAppType
+) => {
+  const appTypeAssets = appAssets.find((aa) => aa.appType === appType);
+  return (
+    <StatusIcon {...{ config, iconConfig }} iconPath={appTypeAssets?.icon}>
+      <AppStatusIcon
+        {...{ workspaceNamespace, userSuspended }}
+        appStatus={status}
+        style={styles.statusIconContainer}
+      />
+    </StatusIcon>
+  );
+};
+
 const displayRuntimeStatusIcon = (
```

Comment:
If possible, I think this should be converted to a React component. Otherwise, it will completely re-render every time components using it update, which can lead to errors.

Converting it should be straightforward: Merge the arguments into a `prop` object, and call the function with JSX syntax.

Reply:
I had not considered the re-rendering case. Thanks for pointing it out. I converted both to components.